cli/internal/cloudcmd/upgrade.go

```diff
@@ -131,32 +142,41 @@ func (u *Upgrader) UpgradeNodeVersion(ctx context.Context, conf *config.Config)
 	if err := u.updateMeasurements(ctx, conf.GetMeasurements()); err != nil {
 		return fmt.Errorf("updating measurements: %w", err)
 	}
-	updatedNodeVersion, err := u.applyUpgrade(ctx, &components, nodeVersion)
+
+	updatedNodeVersion, err := u.applyNodeVersion(ctx, nodeVersion)
 	if err != nil {
 		return fmt.Errorf("applying upgrade: %w", err)
 	}
 	if updatedNodeVersion.Spec.ImageReference != imageReference ||
 		updatedNodeVersion.Spec.ImageVersion != imageVersion.Version ||
-		updatedNodeVersion.Spec.KubernetesComponentsReference != components.ObjectMeta.Name ||
+		updatedNodeVersion.Spec.KubernetesComponentsReference != nodeVersion.Spec.KubernetesComponentsReference ||
 		updatedNodeVersion.Spec.KubernetesClusterVersion != versionConfig.ClusterVersion {
 		return errors.New("unexpected value in updated nodeVersion object")
 	}
 
 	return errors.Join(upgradeErrs...)
 }
 
-func (u *Upgrader) applyUpgrade(ctx context.Context, components *corev1.ConfigMap, nodeVersion updatev1alpha1.NodeVersion) (updatev1alpha1.NodeVersion, error) {
+// applyComponentsCM applies the k8s components ConfigMap to the cluster.
+func (u *Upgrader) applyComponentsCM(ctx context.Context, components *corev1.ConfigMap) error {
+	if components == nil {
+		return errors.New("components ConfigMap is nil")
+	}
```

Comment:
Having dedicated input checks which simply error, is uncommon in Go at least from what I've seen. If the components is nil, the function who actually uses this value should return an error, which we handle i.e. `createConfigMap(ctx, components)`. Furthermore, the `components` passed to this function needed to be created in a function before, which should return either a valid `components` or an error we handle.

Reply:
Had this for more descriptive error messages. Removed. All other aspects you mentioned are already true.